mailbox/jpa/src/main/java/org/apache/james/mailbox/jpa/mail/JPAAttachmentMapper.java

```diff
@@ -0,0 +1,154 @@
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import org.apache.james.mailbox.exception.AttachmentNotFoundException;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.jpa.JPAId;
+import org.apache.james.mailbox.jpa.JPATransactionalMapper;
+import org.apache.james.mailbox.jpa.mail.model.JPAAttachment;
+import org.apache.james.mailbox.model.Attachment;
+import org.apache.james.mailbox.model.AttachmentId;
+import org.apache.james.mailbox.model.MessageId;
+import org.apache.james.mailbox.store.mail.AttachmentMapper;
+import org.apache.james.mailbox.store.mail.model.Username;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.PersistenceException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class JPAAttachmentMapper extends JPATransactionalMapper implements AttachmentMapper {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(JPAAttachmentMapper.class);
+
+
+    public JPAAttachmentMapper(EntityManagerFactory entityManagerFactory) {
+        super(entityManagerFactory);
+    }
+
+
+    @Override
+    public Attachment getAttachment(AttachmentId attachmentId) throws AttachmentNotFoundException {
+        Preconditions.checkArgument(attachmentId != null);
+        return Optional.ofNullable(getEntityManager().find(JPAAttachment.class, attachmentId.getId()))
+                .orElseThrow(() -> new AttachmentNotFoundException(attachmentId.getId())).toAttachment();
+    }
+
+    @Override
+    public List<Attachment> getAttachments(Collection<AttachmentId> attachmentIds) {
+        Preconditions.checkArgument(attachmentIds != null);
+        return ImmutableList.<Attachment>builder().addAll(
+               attachmentIds.stream().distinct()
+               .map(attachmentId -> getEntityManager().find(JPAAttachment.class, attachmentId.getId()))
+                       .filter(Objects::nonNull)
+                       .map(jpaAttachment ->  jpaAttachment.toAttachment())
+               .collect(Collectors.toList())
+
+        ).build();
+
+
+    }
+
+
+    @Override
+    public void storeAttachmentForOwner(Attachment attachment, Username owner) throws MailboxException {
+        Preconditions.checkArgument(owner != null);
+        Preconditions.checkArgument(attachment != null);
+        JPAAttachment jpaAttachment = getJpaAttachment(attachment);
+        try {
+            jpaAttachment.getOwners().add(owner.getValue());
+            getEntityManager().getTransaction().begin();
+            getEntityManager().merge(jpaAttachment);
+            getEntityManager().getTransaction().commit();
+        } catch (PersistenceException e) {
+            LOGGER.warn("Persist attachment for owner met an exception", e);
```

Comment:
You can set error level I think. Logging information about `owner` could be helpful!